From: Bernhard Walle <bernhard@bwalle.de>
Date: Sat, 21 Jan 2012 23:00:43 +0100
Subject: [PATCH] Darwin: Provide dummy implementation of gettext functions

It's just not worth the effort requiring gettext for building locales in
a cross build system. English error messages are sufficient.

Signed-off-by: Bernhard Walle <bernhard@bwalle.de>
---
 include/libintl.h |   23 ++++++++++++++++++++++-
 1 files changed, 22 insertions(+), 1 deletions(-)

diff --git a/include/libintl.h b/include/libintl.h
index 096aa8c..edf34eb 100644
--- a/include/libintl.h
+++ b/include/libintl.h
@@ -1,6 +1,27 @@
 #ifdef HAVE_LIBINTL_H
 #include_next <libintl.h>
-#endif
+#else
+
+#ifndef LIBINTL_H_
+#define LIBINTL_H_  1
+
+// dummy implementations of gettext() and textdomain()
+//
+static inline char *gettext(const char *msgid)
+{
+    return (char *)msgid;
+}
+
+static inline char *textdomain (const char * domainname)
+{
+    static char current_domain[1024];
+    strncpy(current_domain, domainname, sizeof(current_domain));
+    current_domain[sizeof(current_domain)-1] = '\0';
+    return current_domain;
+}
+
+#endif /* LIBINTL_H_ */
+#endif /* !HAVE_LIBINTL_H */
 
 #ifndef _
 #define _(X) (X)
